var/spack/repos/builtin/packages/hipblas/package.py

```diff
@@ -31,6 +31,9 @@ class Hipblas(CMakePackage):
         depends_on('rocsolver@' + ver, when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
+
+    patch('hipblas-remove-opt-rocm-reference.patch', when='@4.2.0:')
```

Comment:
This patch shouldn't be necessary; CMAKE_MODULE_PATH is appended after it searches for rocm-cmake modules, and afterwards the only module being looked for is HIP if USE_CUDA is enabled (which we don't do).

Adding the dependency on rocm-cmake should be enough, spack adds it to CMAKE_PREFIX_PATH, which has higher precedence than the hard-coded PATHS, see https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure